BraveRewards/Ledger/BATBraveLedger.mm

```diff
@@ -757,7 +845,199 @@ - (void)confirmationsTransactionHistoryDidChange
 
 #pragma mark - Notifications
 
-// TODO: Add notification logic here
+- (NSArray<BATRewardsNotification *> *)notifications
+{
+  return [self.mNotifications copy];
+}
+
+- (void)clearNotificationWithID:(NSString *)notificationID
+{
+  for (BATRewardsNotification *n in self.notifications) {
+    if ([n.id isEqualToString:notificationID]) {
+      [self clearNotification:n];
+      return;
+    }
+  }
+}
+
+- (void)clearNotification:(BATRewardsNotification *)notification
+{
+  [self.mNotifications removeObject:notification];
+}
+
+- (void)clearAllNotifications
+{
+  [self.mNotifications removeAllObjects];
+}
+
+- (void)startNotificationTimers
+{
+  if (!self.isEnabled) {
+    return;
+  }
+  
+  // Startup timer, begins after 30-second delay.
+  self.notificationStartupTimer =
+  [NSTimer scheduledTimerWithTimeInterval:30
+                                   target:self
+                                 selector:@selector(checkForNotificationsAndFetchGrants)
+                                 userInfo:nil
+                                  repeats:NO];
+}
+
+- (void)checkForNotificationsAndFetchGrants
+{
+  self.lastNotificationCheckDate = [NSDate date];
+  
+  [self showBackupNotificationIfNeccessary];
+  [self showAddFundsNotificationIfNeccessary];
+  [self fetchGrants:std::string() paymentId:std::string()];
+}
+
+- (void)showBackupNotificationIfNeccessary
+{
+  // This is currently not required as the user cannot manage their wallet on mobile... yet
+  /*
+  auto bootstamp = ledger->GetBootStamp();
+  auto userFunded = [self.prefs[kUserHasFundedKey] boolValue];
+  auto backupSucceeded = [self.prefs[kBackupSucceededKey] boolValue];
+  if (userFunded && !backupSucceeded) {
+    auto frequency = 10; [self.prefs[kBackupNotificationFrequencyKey] doubleValue];
+    auto interval = 10; [self.prefs[kBackupNotificationIntervalKey] doubleValue];
+    auto delta = [[NSDate date] timeIntervalSinceDate:[NSDate dateWithTimeIntervalSince1970:bootstamp]];
+    if (delta > interval) {
+      auto nextBackupNotificationInterval = frequency + interval;
+      self.prefs[kBackupNotificationIntervalKey] = @(nextBackupNotificationInterval);
+      [self savePrefs];
+      [self addNotificationOfKind:BATRewardsNotificationKindBackupWallet
+                        arguments:nil
+                   notificationID:@"rewards_notification_backup_wallet"];
+    }
+  }
+  */
+}
+
+- (void)showAddFundsNotificationIfNeccessary
+{
+  const auto stamp = ledger->GetReconcileStamp();
+  const auto now = [[NSDate date] timeIntervalSince1970];
+  
+  // Show add funds notification if reconciliation will occur in the
+  // next 3 days and balance is too low.
+  if (stamp - now > 3 * 24 * 60) {
```

Comment:
missing one more `60` for minutes here, can make a constant for it, as it's used in two places in this method